app/models/customer.rb

```diff
@@ -1,4 +1,17 @@
 class Customer < ApplicationRecord
    has_many :invoices
+   has_many :transactions, through: :invoices
    validates_presence_of :first_name, :last_name
+
+   def self.top_five_successful_transactions
+    self.joins(:transactions)
+        .where("transactions.result = 1")
```

Comment:
Perhaps refactor this to `where(transactions: {result: 1})` although I'm not sure if that is better or if there is any difference.